bundles/org.openhab.binding.tr064/src/main/java/org/openhab/binding/tr064/internal/phonebook/Tr064PhonebookImpl.java

```diff
@@ -33,37 +33,42 @@
 public class Tr064PhonebookImpl implements Phonebook {
     private final Logger logger = LoggerFactory.getLogger(Tr064PhonebookImpl.class);
 
-    private Map<String, String> phonebook = new HashMap<>();
+    protected Map<String, String> phonebook = new HashMap<>();
 
     private final HttpClient httpClient;
     private final String phonebookUrl;
+    private final int httpTimeout;
 
     private String phonebookName = "";
 
-    public Tr064PhonebookImpl(HttpClient httpClient, String phonebookUrl) {
+    public Tr064PhonebookImpl(HttpClient httpClient, String phonebookUrl, int httpTimeout) {
         this.httpClient = httpClient;
         this.phonebookUrl = phonebookUrl;
+        this.httpTimeout = httpTimeout;
         getPhonebook();
     }
 
     private void getPhonebook() {
-        PhonebooksType phonebooksType = Util.getAndUnmarshalXML(httpClient, phonebookUrl, PhonebooksType.class);
-        if (phonebooksType != null) {
-            phonebookName = phonebooksType.getPhonebook().getName();
-            phonebook = phonebooksType.getPhonebook().getContact().stream().map(contact -> {
-                String contactName = contact.getPerson().getRealName();
-                return contact.getTelephony().getNumber().stream()
-                        .collect(Collectors.toMap(number -> normalizeNumber(number.getValue()), number -> contactName,
-                                this::mergeSameContactNames));
-            }).collect(HashMap::new, HashMap::putAll, HashMap::putAll);
-            logger.debug("Downloaded phonebook {}: {}", phonebookName, phonebook);
+        PhonebooksType phonebooksType = Util.getAndUnmarshalXML(httpClient, phonebookUrl, PhonebooksType.class,
+                httpTimeout);
+        if (phonebooksType == null) {
+            logger.warn("Failed to get phonebook with URL '{}'", phonebookUrl);
+            return;
         }
+        phonebookName = phonebooksType.getPhonebook().getName();
+
+        phonebook = phonebooksType.getPhonebook().getContact().stream().map(contact -> {
+            String contactName = contact.getPerson().getRealName();
+            return contact.getTelephony().getNumber().stream().collect(Collectors.toMap(
+                    number -> normalizeNumber(number.getValue()), number -> contactName, this::mergeSameContactNames));
+        }).collect(HashMap::new, HashMap::putAll, HashMap::putAll);
+        logger.debug("Downloaded phonebook {}: {}", phonebookName, phonebook);
     }
 
     // in case there are multiple phone entries with same number -> name mapping, i.e. in phonebooks exported from
     // mobiles containing multiple accounts like: local, cloudprovider1, messenger1, messenger2,...
     private String mergeSameContactNames(String nameA, String nameB) {
-        if (nameA.equals(nameB)) {
+        if (nameA != null && nameA.equals(nameB)) {
```

Comment:
This was recently fixed, please revert. 
`nameA` cannot be null

Reply:
I'm not 100% sure this is correct, but in case the name is `null` the check must be done at another place. Please see my fix.

Reply:
@lsiepel, maybe you can recall from #13873?

Reply:
The class is `@NonNullByDefault` so atleast in this method it can't be null. 

It comes from => contact.getPerson().getRealName() (line 61) that calls PersonType.getRealName() that gets it from XML node realName that has attribute requireed = true

The compiler also warns: Redundant null check: comparing '@NonNull String' against null

So i guess this really can;t be null ;-) but, its just a null check, if still unsure, let's keep it. It should certainly not block this mayor update to TR064 binding

Reply:
@NonNullByDefault is a compile-time check. That’s why the compiler complains: IF the annotation is correct, THEN the null-check is not needed.

OTOH, there is nothing preventing a null value coming here at runtime. IIRC the I have written the XSD for that by hand, so I would not bet on it being correct in all aspects. Maybe we can just try if it is possible to add a phonebook contact in the FritzBox with a blank name and how that looks like in the XML.